frmts/vrt/pixelfunctions.cpp

```diff
@@ -59,6 +59,8 @@ inline double GetSrcVal(const void *pSource, GDALDataType eSrcType, T ii)
         case GDT_Int64:
             return static_cast<double>(
                 static_cast<const int64_t *>(pSource)[ii]);
+        case GDT_Float16:
```

Comment:
The data type enumeration in frmts/vrt/data/gdalvrt.xsd will need to be extended

Reply:
Done. I assume you refer to `DataTypeType`. I assume the datatypes there are essentially unordered and do not have to have the same order as `enum GDALDataType`. I also added `Int8` which seemed to be missing.